punchcut/src/lib.rs

```diff
@@ -0,0 +1,191 @@
+/*!
+Glyph loading.
+*/
+
+#![forbid(unsafe_code)]
+// TODO: this is temporary-- remove when hinting is added.
+#![allow(dead_code, unused_imports, unused_variables)]
+
+/// Re-export of peniko crate.
+pub use peniko;
+
+mod error;
+mod scaler;
+mod sink;
+
+pub mod source;
+
+/// Representations of fonts and font collections.
+pub mod font {
+    pub use read_fonts::{types::Tag, CollectionRef, FileRef, FontRef, TableProvider};
+}
+
+use font::Tag;
+use source::glyf;
+
+use core::str::FromStr;
+
+pub use error::{Error, Result};
+pub use scaler::{Scaler, ScalerBuilder};
+pub use sink::PathSink;
+
+/// Modes for hinting.
+///
+/// Only the `glyf` source supports all hinting modes.
+#[derive(Copy, Clone, PartialEq, Eq, Default, Debug)]
+#[allow(dead_code)]
+pub enum Hinting {
+    /// "Full" hinting mode. May generate rough outlines and poor horizontal
+    /// spacing.
+    Full,
+    /// Light hinting mode. This prevents most movement in the horizontal direction
+    /// with the exception of a per-font backward compatibility opt in.
+    Light,
+    /// Same as light, but with additional support for RGB subpixel rendering.
+    LightSubpixel,
+    /// Same as light subpixel, but always prevents adjustment in the horizontal
+    /// direction. This is the default mode.
+    #[default]
+    VerticalSubpixel,
+}
+
+/// Type for a normalized variation coordinate.
+pub type NormalizedCoord = read_fonts::types::F2Dot14;
+
+/// Type for a glyph identifier.
+pub type GlyphId = read_fonts::types::GlyphId;
+
+/// Setting for specifying a variation by tag and value.
+#[derive(Copy, Clone, Debug)]
+pub struct Variation {
+    /// Tag for the variation.
+    pub tag: Tag,
+    /// Value for the variation.
+    pub value: f32,
+}
+
+impl From<(Tag, f32)> for Variation {
+    fn from(s: (Tag, f32)) -> Self {
+        Self {
+            tag: s.0,
+            value: s.1,
+        }
+    }
+}
+
+impl From<(&str, f32)> for Variation {
+    fn from(s: (&str, f32)) -> Self {
+        Self {
+            tag: Tag::from_str(s.0).unwrap_or_default(),
+            value: s.1,
+        }
+    }
+}
+
+impl From<([u8; 4], f32)> for Variation {
+    fn from(s: ([u8; 4], f32)) -> Self {
+        Self {
+            tag: Tag::new_checked(&s.0[..]).unwrap_or_default(),
+            value: s.1,
+        }
+    }
+}
+
+/// Context for loading glyphs from any available source.
+#[derive(Default)]
+pub struct Context {
+    /// Inner context for loading TrueType outlines.
+    glyf: glyf::Context,
+    /// Internal storage for TrueType outlines.
+    glyf_outline: glyf::Outline,
```

Comment:
It says above it's for any source but it appears to only support glyf. If "any source" is aspirational comment the type so and explain how you imagine extending it. For example, would the glyf context and glyf become an enum?

Reply:
The context is font independent so it would grow support for other formats (CFF, COLR) with additional fields. I'll add a note.